File: web-interface/tpl/json/xivo/configuration/manage/entity.php
<?php

#
# XiVO Web-Interface
# Copyright (C) 2009  Proformatique <technique@proformatique.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

dwho::load_class('dwho_http');
$http_response = dwho_http::factory('response');

if($this->get_var('act') === 'view')
	$data = dwho_json::encode($this->get_var('info'));
else
{
	$list = $this->get_var('list');

	if(is_array($list) === false)
	{
		$http_response->set_status_line(500);
		$http_response->send(true);
	}
	else if(($nb = count($list)) === 0)
	{
		$http_response->set_status_line(204);
		$http_response->send(true);
	}

	$data = array();

	for($i = 0;$i < $nb;$i++)
		$data[] = $list[$i]['entity'];

	$data = dwho_json::encode($data);
}

if($data === false)
{
	$http_response->set_status_line(500);
	$http_response->send(true);
}

$sum = $this->get_var('sum');

if(isset($sum{0}) === true && $sum === md5($data))
{
	$http_response->set_status_line(304);
	$http_response->send(true);
}

header(dwho_json::get_header());
die($data);

?>

File: web-interface/unittest/xivojson/entity.py
# -*- coding: utf-8 -*-
from __future__ import with_statement

__version__ = "$Revision$ $Date$"
__author__  = "Guillaume Bour <gbour@proformatique.com>"
__license__ = """
    Copyright (C) 2010  Proformatique

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA..
"""
import unittest, pprint, cjson
from xivojson import *
# define global variables
from globals  import *


class Test01Entity(unittest.TestCase):
    def setUp(self):
        global IP, PORT, SSL, USERNAME, PASSWORD

        self.client = JSONClient(IP, PORT, SSL, USERNAME, PASSWORD)

    def test_01_entity(self):
        (resp, data) = self.client.list('entity')
#        print resp.status
#        pprint.pprint(data)

